otherlibs/stdune/src/io.ml

```diff
@@ -268,22 +268,28 @@ struct
       fun ?chmod ~src ~dst () ->
         let src = Path.to_string src in
         let dst = Path.to_string dst in
-        let src_stats = Unix.stat src in
-        (match src_stats.st_kind with
-        | S_DIR -> raise (Sys_error "Is a directory")
-        | _ -> ());
-        (try Copyfile.copyfile src dst with
-        | Unix.Unix_error (Unix.EPERM, "unlink", _) ->
-          let message = Printf.sprintf "%s: Is a directory" dst in
-          raise (Sys_error message)
-        | Unix.Unix_error (Unix.ENOENT, "realpath", _) ->
+        match Unix.stat src with
```

Comment:
Looks fine, but I prefer to avoid nesting whenever possible:

```
let stats =
  match Unix.stat src with
  | s -> s
  | exception .. -> .. raise
in
```

Reply:
Ok, I can do that

Reply:
Removed nesting in latest push.